src/odo.ts

```diff
@@ -162,9 +162,17 @@ export class OdoImpl implements Odo {
         const result: cliInstance.CliExitData = await this.execute(
             `oc get dc --namespace ${proj} -o jsonpath="{range .items[?(.metadata.labels.app == \\"${application.getName()}\\")]}{.metadata.labels.app\\.kubernetes\\.io/component-name}{\\"\\n\\"}{end}"`
         );
-        return result.stdout.trim().split('\n')
+
+        const componentsList = result.stdout.trim().split('\n')
             .filter((value) => value !== '')
             .map<OpenShiftObject>((value) => new OpenShiftObjectImpl(application, value, 'component', this, TreeItemCollapsibleState.Collapsed));
+        if (componentsList.length>0) {
```

Comment:
you can do simply:
``` JavaScript
commands.executeCommand('setContext', 'componentPresent', componentsList.length>0);
```
I would also add  `twoComponentsPresent' to block  `openshift.component.link` action if only one component is available, but that is optional.

Reply:
> you can do simply:
> 
> ```js
> commands.executeCommand('setContext', 'componentPresent', componentsList.length>0);
> ```

Fixed.